tests/Functional/Schema/AlterTableTest.php

```diff
@@ -0,0 +1,228 @@
+                    . ' which is a bug.',
+            );
+        }
+
+        $this->ensureDroppingPrimaryKeyConstraintIsSupported();
+
+        $table = new Table('alter_pk');
+        $table->addColumn('id1', Types::INTEGER, ['autoincrement' => true]);
+        $table->addColumn('id2', Types::INTEGER);
+        $table->setPrimaryKey(['id1', 'id2']);
+
+        $this->testMigration($table, static function (Table $table): void {
+            $table->dropPrimaryKey();
+            $table->setPrimaryKey(['id1']);
+        });
+    }
+
+    /** @throws Exception */
+    public function testAddNonAutoincrementColumnToPrimaryKeyWithAutoincrementColumn(): void
+    {
+        $platform = $this->connection->getDatabasePlatform();
+
+        if ($platform instanceof AbstractMySQLPlatform) {
+            self::markTestIncomplete(
+                'DBAL does not restore the auto-increment attribute after dropping and adding the constraint,'
+                    . ' which is a bug.',
+            );
+        }
+
+        if ($platform instanceof SQLitePlatform) {
+            self::markTestSkipped(
+                'SQLite does not support auto-increment columns as part of composite primary key constraint',
+            );
+        }
+
+        $this->ensureDroppingPrimaryKeyConstraintIsSupported();
+
+        $table = new Table('alter_pk');
+        $table->addColumn('id1', Types::INTEGER, ['autoincrement' => true]);
+        $table->addColumn('id2', Types::INTEGER);
+        $table->setPrimaryKey(['id1']);
+
+        $this->testMigration($table, static function (Table $table): void {
+            $table->dropPrimaryKey();
+            $table->setPrimaryKey(['id1', 'id2']);
+        });
+    }
+
+    /** @throws Exception */
+    public function testAddNewColumnToPrimaryKey(): void
+    {
+        $this->ensureDroppingPrimaryKeyConstraintIsSupported();
+
+        $table = new Table('alter_pk');
+        $table->addColumn('id1', Types::INTEGER);
+        $table->setPrimaryKey(['id1']);
+
+        $this->testMigration($table, static function (Table $table): void {
+            $table->addColumn('id2', Types::INTEGER);
+            $table->dropPrimaryKey();
+            $table->setPrimaryKey(['id1', 'id2']);
+        });
+    }
+
+    /** @throws Exception */
+    private function ensureDroppingPrimaryKeyConstraintIsSupported(): void
+    {
+        $platform = $this->connection->getDatabasePlatform();
+
+        if (
+            ! ($platform instanceof DB2Platform)
+            && ! ($platform instanceof OraclePlatform)
+            && ! ($platform instanceof SQLServerPlatform)
+        ) {
+            return;
+        }
+
+        self::markTestIncomplete(
+            'Dropping primary key constraint on the currently used database platform is not implemented.',
+        );
+    }
+
+    /** @throws Exception */
+    private function testMigration(Table $oldTable, callable $migration): void
```

Comment:
We should not use the `test` prefix on methods that we don't want PHPUnit to pick up as tests. I am aware that PHPUnit will skip the method because it's private, but still: we can probably find a better name.

Reply:
I don't think using the `test` prefix is a problem right because PHPUnit will skip it, and in this specific case, the method does contain the body of the test. I'm open to suggestions on renaming the method.

Reply:
I'm going to merge as it's not a blocker. Happy to rename this one and potentially other similar methods in a separate PR.